app/controllers/authentications_controller.rb

```diff
@@ -5,6 +5,12 @@ def schoology
   end
 
   def google
+    if(request.params["state"].present? &&
+       request.params["state"].match(/after_sign_in_path=(.*)/))
```

Comment:
Isn't this regexp a bit too loose? I guess it's a theoretical issue, but what if there are more variables in the state?

Reply:
Yeah it is loose. I didn't really define a separator character between parameters other than a space char between the previous state and the after_sign_in_path.   I figure the path might have an unescaped space in it, so unless we escape the path it isn't safe to limit the regex.  I'll add a comment about this to clarify.